src/tool/subcommands/api_cmd.rs

```diff
@@ -204,6 +205,54 @@ impl EndpointStatus {
         }
     }
 }
+
+/// Data about a failed test. Used for debugging.
+struct TestDump {
+    request: RpcRequest,
+    forest_response: Option<String>,
+    lotus_response: Option<String>,
+}
+
+impl std::fmt::Display for TestDump {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        writeln!(f, "Request: {:?}", self.request)?;
```

Comment:
I'd prefer JSON format for requests, which I could copy & paste into tools like postman directly.

Reply:
Yeah, I also prefer it, but avoided it because the conversion looked pretty convoluted and several levels of abstraction away from this, in https://github.com/ChainSafe/forest/blob/state-miner-recoveries-flaky/src/rpc/client.rs#L61-L95... and then it's still converted further in the `jsonrpsee` crate. The point is, on the client side, we don't get the raw JSON request without some massive changes. At this point it'd be up to the one debugging it to enable `jsonrpsee` tracing logs or some other on the node side.

Best I can do is represent params as JSON instead of the Rust `Debug` form. What do you think?

Reply:
I see, there is `trace_params` when we turn on tracing log.  At this request level, it will be great to have all fields than only params so I think the current format is good. Thanks!

Reply:
But at the same time, could we add `forest_filecoin::rpc=trace`, sometimes it's hard to convert the DEBUG format of the params into JSON

Reply:
@hanabi1224 It'd be nice, but at the same it would flood our logs. Just setting it to `debug` ends up in massive log files.

```
❯ docker compose logs api-compare | wc -c | numfmt --to=iec
148M
```

Reply:
How about `writeln!(f, "Request: {:?}, params: {json}", self.request)?;` then?

Reply:
That's what I proposed in the first comment. :laughing:

Reply:
```
api-compare-1  | Request dump: RpcRequest { method_name: "Filecoin.StateMinerActiveSectors", params: Array [String("f03751"), Array [Object {"/": String("bafy2bzacecdk2jtrs5zwlavzoz4ikjov6wr26xghsvx7jgx7buz7lplglqqjq")}, Object {"/": String("bafy2bzacecfup6rqypio77bhr53qxhgypjtq26mpbqwbd4munwgkieoh2lqrm")}, Object {"/": String("bafy2bzacebg3xchktxrivtxhcumxqio3nysqcovzgcwpyqb2nvhzdwv4vk2mw")}]], result_type: PhantomData<serde_json::value::Value>, api_version: V0, timeout: 60s }
api-compare-1  | Request params JSON: ["f03751",[{"/":"bafy2bzacecdk2jtrs5zwlavzoz4ikjov6wr26xghsvx7jgx7buz7lplglqqjq"},{"/":"bafy2bzacecfup6rqypio77bhr53qxhgypjtq26mpbqwbd4munwgkieoh2lqrm"},{"/":"bafy2bzacebg3xchktxrivtxhcumxqio3nysqcovzgcwpyqb2nvhzdwv4vk2mw"}]]
```
Is this the droid you were looking for? @hanabi1224

Reply:
Yes, that's perfect, thanks!